specs/Deployment/DeploymentAPI.md

```diff
@@ -3,101 +3,117 @@
+This is the API an app or program can call to verify that all required WinAppSDK packages are
+installed to a minimum version needed by the Framework loaded with the app and that the packages are
+in good state. A process should call GetStatus() after process initialization and before using
+Windows App Runtime content (e.g. in Main() or WinMain()).
 
 Developers call this method without parameters, as all information about the version, channel, and
-architecture are derived from the current framework package loaded.
+architecture are derived from the current Framework package loaded.
 
 ```C#
     DeploymentResult result = DeploymentManager.GetStatus();
     if (result.Status == DeploymentStatus.Ok)
     {
-        // WindowsAppRuntime is in a good state and ready for use.
+        // WindowsAppRuntime is installed and is in a good state and ready for use.
     }
     else
     {
-        // WindowsAppRuntime is not in a good state.
-        // Should call Initialize on another thread, or not use features that require the
-        // WindowsAppRuntime. result.ExtendedError has the error code that has the reason
-        // for the Status.
+        // WindowsAppRuntime is either not installed or not in a good state (can't be both at same time).
+        if (result.Status == DeploymentStatus.PackageInstallRequired)
+        {
+            // Should call Initialize on another thread, or not use features that require the
+            // WindowsAppRuntime. result.ExtendedError has the error code that has the reason
+            // for the Status.
+        }
+        else if (result.Status == DeploymentStatus.PackageRepairRequired ||
+                result.Status == DeploymentStatus.Unknown)
+        {
+            // Should call Repair on another thread, or not use features that require the
+            // WindowsAppRuntime. result.ExtendedError has the error code that has the reason
+            // for the Status.
+        }
     }
 ```
 
 ## Initialize
 
-This API performs a status check, and if the main and/or singleton packages are missing or not at
+This API performs a status check, and if the Main and/or Singleton packages are missing or not at
 the required version, it will attempt to deploy those packages and get the WindowsAppRuntime into an
 OK state. All information about the version, channel, and architecture needed are derived from the
-current WinAppSDK framework package.
-
-Since both the main _and_ the singleton packages may need to be installed, it is possible that one
-package is installed successfully but not the other. If so, the returned WindowsAppRuntimeStatus
-will contain the error, if one occurs during package install. There will be no rollback of any
-successfully installed packages. See the
+current WinAppSDK Framework package.
+
+Since both the Main _and_ the Singleton packages may need to be installed, it is possible that Main
+package install has failed and Initialize returns with PackageInstallFailed state (OR) it is
```

Comment:
What happens if I try to Repair() a package that's in PackageInstallFailed state?  That seems like a natural inclination ("I did Initialize() and it's in some error state now, I guess that means it's broken so I should fix it. Fix -> repair. Call Repair()").  Are we providing DeploymentResult properties that tell the developer they called the wrong method?

Reply:
Repair() essentially provides an unconditional core functionality of Initialize() - which is to install WindowsAppSDK Main and Singleton licenses and packages. So, if Repair() is tried when Initialize() fails, then it is effectively a retry of Initialize(). If the underlying issue for PackageInstallFailed is transient, then Repair() may succeed. Otherwise, it will continue to be PackageInstallFailed.